### PR TITLE
App Store Templates

### PR DESCRIPTION
**Pull request by hariombalhara**
_NOTE: Many of the files are just tiny modifications_

## Why Templates
A template is a high-quality example app that can be used as a starting point for different types of apps.

## How to use a Template
`yarn create-app -t TEMPLATE_NAME` 

Example: `yarn create-app -t basic`

If the base template isn't provided in the argument, it can be chosen later on as well.
Screenshot 2023-01-13 at 9 13 03 PM

## How to create a  new Template

`yarn create-app-template -t TEMPLATE_NAME` 

Example: `yarn create-app-template -t basic`

If the base template isn't provided in the argument, it can be chosen later on as well in the same way it can be selected while creating an app.

## Available Templates

_Templates are available in the dev environment only for installation so that they can be tested end to end._
_My Template is a test template created using `create-app-template` command_
Screenshot 2023-01-13 at 9 17 35 PM

All templates exist in `@calcom/app-store/templates`
### Basic
It is a template for an app that can be installed and provides no other feature.

###  Booking Pages Tag
It is a template demoing a Booking Pages tracking app like GA4, Fathom and Plausible.

### EventType AppCard
A template showcasing how an app with EventType AppCard can be built e.g. Giphy, QR Code

### EventType Location Video - Static
 It is a template showing how to add a static URL EventType location e.g. Around, Whereby

### General App Settings
 It is a template showing how an App can have settings in installed section that configures the app globally e.g. Weather in your Calendar

### Link as An App
It's a template showing how an app, that is just a link to some webpage, can be made e.g. Pipedream, Amie, Vimcal

### New Commands
- `yarn app-store create-template`
- `yarn app-store edit-template`
- `yarn app-store delete-template`

### More User-friendly aliases
- `yarn create-app` -> `yarn app-store create`
- `yarn edit-app` -> `yarn app-store edit`
- `yarn delete-app` -> `yarn app-store delete`
- `yarn create-app-template` -> `yarn app-store create-template`
- `yarn edit-app-template` -> `yarn app-store edit-template`
- `yarn delete-app-template `-> `yarn app-store delete-template`

## Notable Changes
- [x] Renamed `EventTypeAppCard.tsx`  to `EventTypeAppCardInterface.tsx`. Interface suffix at the end conveys that this is a special component(interface b/w the App Code and AppStore) that AppStore would read and its name can't be changed.
- [x] **Removed _metadata.ts from templates and thus new apps won't create it.**
- [x] DESCRIPTION.md 
	- [x] Assets can be specified relative to the `static` folder.
	- [x] {DESCRIPTION} template variable is available to directly inject app's description at any place.

## Checklist
- Moved it from `extensions` to `components` folder to avoid introducing a new folder.
- [x] `logo` and `imageSrc` can be specified as relative to `static` 
- [ ] Figure out and remove one of the stale ones in `logo` and `imageSrc`
- [ ] Add zod validation of config.json
- [x] We should be able to debug templates in the dev environment under [localhost:3000/apps/templates](http://localhost:3000/apps/templates) - Shouldn’t be available elsewhere.
- [x] We should ensure that templates are seeded only in Dev Environment. Production environments have no need for it.
- [x] EventTypeAnalytics still requires manual change - Move the tag config to config.json and avoid manual file creation.
- [x] Cli code improvement and UX improvement. 
	- [x]  Got the loader(....) working by using `exec` instead of `execSync`
	- [x]  Templates can be picked from a select if not specified using -t
	- [x]  Throw an error if the specified template is not found.
- [x] Make sure that a template isn't seeded in production.
- [x] Make sure that in Dev mode if the template is seeded and when it's visible in UI, it has an appropriate badge to convey that it is visible only in Dev.

### CODE REVIEW

**Comment by hariombalhara** [comment]
It needs a good amount of work before being opened for Review.

**Review by hariombalhara — COMMENTED** [review]
Self review done

File: packages/prisma/seed-app-store.ts [hunk 1]
```diff
@@ -293,7 +294,17 @@ export default async function main() {
   );
   for (let i = 0; i < generatedApps.length; i++) {
     const generatedApp = generatedApps[i];
-    await createApp(generatedApp.slug, generatedApp.dirName, generatedApp.categories, generatedApp.type);
+    if (generatedApp.isTemplate && process.argv[2] !== "seed-templates") {
```
Comment (hariombalhara) [inline packages/prisma/seed-app-store.ts:297]: It ensures that templates are seeded only if `seed-templates` argument is provided, which prevents templates from ever being seeded into production

File: packages/app-store/qr_code/api/add.ts [hunk 2]
```diff
@@ -4,7 +4,6 @@ import { createDefaultInstallation } from "../../_utils/installation";
 import appConfig from "../config.json";
 
 const handler: AppDeclarativeHandler = {
-  // Instead of passing appType and slug from here, api/integrations/[..args] should be able to derive and pass these directly to createCredential
```
Comment (hariombalhara) [inline packages/app-store/qr_code/api/add.ts:7]: Removed this comment as it's not trivial to do it. So, we will keep it as is for now.

File: packages/app-store/plausible/config.json [hunk 3]
```diff
@@ -11,6 +11,16 @@
   "publisher": "Cal.com, Inc.",
   "email": "help@cal.com",
   "extendsFeature": "EventType",
+  "tag": {
+    "scripts": [
+      {
+        "src": "https://plausible.io/js/script.js",
+        "attrs": {
+          "data-domain": "{TRACKED_DOMAIN}"
+        }
```
[the remark this answers is not in the document]
Comment (hariombalhara) [reply packages/app-store/plausible/config.json:20]: Plausible doesn't work on production as well. Opened an issue to track it.

**Comment by PeerRich** [comment]
what a huge but exciting feature! amazing work. @calcom/reviewers lets get this reviewed 🎉

**Review by zomars — APPROVED** [review]
Awesome work @hariombalhara 🙏🏽 

I got no objections so far. Tested locally and works as intended.

**Comment by shirazdole** [comment]
* not a high priority - only answer if easy and short (less than 2 mins) * 
@hariombalhara - what's the best way for me to explain this to engineers on a sales call? Typically, I'll say you can add an app to our app store and send them the 3rd party app docs. How do I include this in my pitch? "We have templates available," but I don't get how the templates work from this PR. I believe this is the most important portion, but I'm too stupid to understand: 

Basic
It is a template for an app that can be installed and provides no other feature.

Booking Pages Tag
It is a template demoing a Booking Pages tracking app like GA4, Fathom and Plausible.

EventType AppCard
A template showcasing how an app with EventType AppCard can be built e.g. Giphy, QR Code

EventType Location Video - Static
It is a template showing how to add a static URL EventType location e.g. Around, Whereby

General App Settings
It is a template showing how an App can have settings in installed section that configures the app globally e.g. Weather in your Calendar

Link as An App
It's a template showing how an app, that is just a link to some webpage, can be made e.g. Pipedream, Amie, Vimcal

**Pull request merged**